Add defaults.go and related tests

robertkotcher opened the pull request.

Commit:
Require go 1.13 in go.mod

Commit:
Add defaults.go and related tests

defaults.go introduces a function to Schema, named InsertDefaults, that
applies default values defined in the schema to an incoming interface

svrana commented on go.mod:
```diff
@@ -1,5 +1,7 @@
 module github.com/xeipuuv/gojsonschema
 
+go 1.14
```
doesn't seem to match your commit message. why do you need to change it?

robertkotcher replied:
You must have commented as I was updating. This is now out of date

svrana reviewed (commented):
be great to leave a real commit message

robertkotcher commented:
> be great to leave a real commit message

@svrana If by "a real commit message" you are requesting for me to elaborate more on what exactly defaults.go does, please say something like that instead.

svrana commented:
Did you get this code from somewhere? Please link to original patchset or somehow credit the original author if there is one.

Commit:
Add support for arrays of types

If type is an array, we currently expect that one of the values in the
array is null, and the other is either "object" or "array".

It would certainly be possible to expand support for more intricate type
defs, but I'd suggest that we wait until we encounter those cases before
supporting them.

See discussion above func primaryType in defaults.go
